src/odemis/acq/test/millmng_test.py

```diff
@@ -0,0 +1,351 @@
+                                "estimated time is less than the actual time while milling the second feature")
+
+        total_est_time_without_drift = task.estimate_milling_time(site_idx=0)
+
+        # B. With Drift
+        milling_setting_1 = MillingSettings(name='rough_milling_1', current=20e-12, horizontal_fov=35e-6,
+                                            roi=(0.5, 0.5, 0.8, 0.8),
+                                            pixel_size=(3.41796875e-08, 3.41796875e-08), beam_angle=self.beam_angle, duration=20,
+                                            dc_roi=(0, 0.3, 0.4, 0.7), dc_period=10, dc_dwell_time=10e-6,
+                                            dc_current=20e-12)
+
+        milling_setting_2 = MillingSettings(name='rough_milling_2', current=20e-12, horizontal_fov=35e-6,
+                                            roi=(0.5, 0.5, 0.8, 0.8),
+                                            pixel_size=(3.41796875e-08, 3.41796875e-08), beam_angle=self.beam_angle, duration=20,
+                                            dc_roi=(0, 0.3, 0.4, 0.7), dc_period=10, dc_dwell_time=10e-6,
+                                            dc_current=20e-12)
+
+        millings = [milling_setting_1, milling_setting_2]
+
+        task = MillingRectangleTask(model.InstantaneousFuture(), millings, self.sites, self.feature_post_status,
+                                    self.acq_streams, self.ion_beam,
+                                    self.sed, self.stage, self.aligner)
+
+        # Time estimate in the beginning
+        time_estimated = task.estimate_milling_time(site_idx=0)
+        drift_estimation_1 = AnchoredEstimator(self.ion_beam, self.sed,
+                                               milling_setting_1.dcRoi.value, milling_setting_1.dcDwellTime.value, max_pixels=512 ** 2, follow_drift=False)
+        drift_estimation_2 = AnchoredEstimator(self.ion_beam, self.sed,
+                                               milling_setting_2.dcRoi.value, milling_setting_2.dcDwellTime.value, max_pixels=512 ** 2, follow_drift=False)
+        min_drift_time = drift_estimation_1.estimateAcquisitionTime() + drift_estimation_2.estimateAcquisitionTime()
+
+        total_time_with_drift = milling_setting_1.duration.value + milling_setting_2.duration.value + min_drift_time
+        self.assertGreaterEqual(time_estimated, total_time_with_drift,
+                                "estimated time with drift correction is less than the minimum time set to mill two features")
+
+        # check milling time without drift correction is less than milling with drift correction
+        total_est_time_with_drift = task.estimate_milling_time(site_idx=0)
+        self.assertLessEqual(total_est_time_without_drift, total_est_time_with_drift,
+                             "milling time with drift correction is less than milling time without drift correction")
+
+    @patch.object(odemis.acq.orsay_milling.OrsayMilling, 'do_milling', fake_do_milling)
```

Comment:
Ideally the patch should be done only if TEST_NOHW. To do so, you probably can place them in setUp() and tearDown()... but I don't know exactly how ;-)